pkg/cloudproviders/gcp/cred_manager_impl.go

```diff
@@ -0,0 +1,76 @@
+package gcp
+
+import (
+	"context"
+
+	"github.com/stackrox/rox/pkg/secretinformer"
+	"github.com/stackrox/rox/pkg/sync"
+	"golang.org/x/oauth2/google"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+const (
+	cloudCredentialsKey = "credentials"
+)
+
+type gcpCredentialsManagerImpl struct {
+	informer  *secretinformer.SecretInformer
+	stsConfig []byte
+	mutex     sync.RWMutex
+}
+
+var _ CredentialsManager = &gcpCredentialsManagerImpl{}
+
+// NewCredentialsManager creates a new GCP credential manager.
+func NewCredentialsManager(k8sClient kubernetes.Interface, namespace string, secretName string) *gcpCredentialsManagerImpl {
+	mgr := &gcpCredentialsManagerImpl{}
+	mgr.informer = secretinformer.NewSecretInformer(
+		namespace,
+		secretName,
+		k8sClient,
+		mgr.updateSecret,
+		mgr.updateSecret,
+		mgr.deleteSecret,
+	)
+	return mgr
+}
+
+func (c *gcpCredentialsManagerImpl) updateSecret(secret *v1.Secret) {
+	if stsConfig, ok := secret.Data[cloudCredentialsKey]; ok {
+		c.mutex.Lock()
+		defer c.mutex.Unlock()
+		c.stsConfig = []byte(stsConfig)
```

Comment:
No need for conversion since we already have byte:
```suggestion
		c.stsConfig = stsConfig
```